src/app/drive/components/ShareInviteDialog/ShareInviteDialog.tsx

```diff
@@ -0,0 +1,231 @@
+  const emailRegex =
+    /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/;
+  const email = watch('email');
+  const [userRole, setUserRole] = useState<string>('editor');
+  const [usersToInvite, setUsersToInvite] = useState<Array<UsersToInvite>>([]);
+  const [notifyUser, setNotifyUser] = useState<boolean>(false);
+  const [messageText, setMessageText] = useState<string>('');
+  const [isInviteButtonDisabled, setIsInviteButtonDisabled] = useState<boolean>(true);
+
+  useEffect(() => {
+    if (email) {
+      isValidEmail(email) || usersToInvite.length > 0
+        ? setIsInviteButtonDisabled(false)
+        : setIsInviteButtonDisabled(true);
+    }
+  }, [email]);
+
+  const onAddInviteUser: SubmitHandler<IFormValues> = (formData, event) => {
+    event?.preventDefault();
+    const userInvitedEmail = formData.email;
+    const userInvitedRole = formData.userRole;
+    const userInvited = { email: userInvitedEmail, userRole: userInvitedRole };
+    const isDuplicated = usersToInvite.find((user) => user.email === userInvited.email);
+
+    if (!isDuplicated) {
+      const unique: Array<UsersToInvite> = [...usersToInvite];
+      unique.push(userInvited);
+      setUsersToInvite(unique);
+    }
+    resetField('email');
+  };
+
+  const onEditRole = () => {
+    //   Edit user role
+  };
+
+  const onInvite = () => {
+    // Ivite added users
+  };
+
+  return (
+    <div>
+      <div className="flex h-16 w-full items-center justify-between space-x-5 border-b border-gray-10 px-5">
+        <h4
+          className="flex max-w-full items-center space-x-4 overflow-hidden overflow-ellipsis whitespace-nowrap text-xl font-medium"
+          title={translate('modals.shareModal.invite.title')}
+        >
+          <ArrowLeft onClick={props.onInviteUser} size={22} className="cursor-pointer" />
+          <span>{translate('modals.shareModal.invite.title')}</span>
+        </h4>
+      </div>
+
+      <div className="p-5">
+        <form className="m flex w-full" onSubmit={handleSubmit(onAddInviteUser)}>
+          <input
+            className="no-ring semi-dense mr-2 w-full flex-grow border border-neutral-30"
+            placeholder={translate('form.fields.email.placeholder') as string}
+            type="text"
+            {...register('email', {
+              required: true,
+              pattern: emailRegex,
+            })}
+          />
+          <Listbox value={userRole} onChange={setUserRole}>
+            {({ open }) => (
+              <div className="relative">
+                <Listbox.Button value={userRole} {...register('userRole')} name="userRole">
+                  <Button variant="secondary">
+                    <span className="capitalize">{userRole}</span>
+                    <CaretDown
+                      size={24}
+                      className={`${open ? 'rotate-180 transform transition' : 'rotate-0 transform transition'}`}
```

Comment:
There should be no rotation animations in the dropdowns (here and in the user list)

Reply:
Done